### ISSUE TITLE
[Bug]: aws-node-termination-handler does not create unique aws_cloudwatch_event_rules

### ISSUE THREAD

**Issue by caleb-devops** [aws-ia/terraform-aws-eks-blueprints#708]
### Welcome to Amazon EKS Blueprints!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/aws-ia/terraform-aws-eks-blueprints/issues) and didn't find any.

### Amazon EKS Blueprints Release version

v4.2.1

### What is your environment, configuration and the example used?

Multiple clusters with `enable_aws_node_termination_handler = true` set.

### What did you do and What did you see instead?

All clusters using the aws-node-termination-handler addon share the same aws_cloudwatch_event_rules. If tags on the aws_cloudwatch_event_rules are modified, or if a cluster is destroyed, all clusters that utilize those event rules will be impacted.

```
module.eks_blueprints_kubernetes_addons.module.aws_node_termination_handler[0].aws_cloudwatch_event_rule.aws_node_termination_handler_rule[0] will be updated in-place
  ~ resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_rule" {
        id             = "NTHSpotTermRule"
        name           = "NTHSpotTermRule"
      ~ tags           = {
          - "Environment" = "Development" -> null
        }
      ~ tags_all       = {
          ~ "Environment"     = "Development" -> "Stage"
            # (4 unchanged elements hidden)
        }
        # (4 unchanged attributes hidden)
    }
```

Fix: Use `name_prefix` instead of `name` in `aws_cloudwatch_event_rule`.
```hcl
resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_rule" {
  count = length(local.event_rules)

  name_prefix   = local.event_rules[count.index].name
  event_pattern = local.event_rules[count.index].event_pattern
}
```

**Comment by Zvikan** [comment]
@askulkarni2 I believe this is same issue as we discussed before when I saw during destroy between karpenter and self-managed node groups:
```
Error: error deleting EventBridge Rule (NTHRebalanceRule): ValidationException: Rule can't be deleted since it has targets.
	status code: 400, request id: <masked>
Error: error deleting EventBridge Rule (NTHInstanceStateChangeRule): ValidationException: Rule can't be deleted since it has targets.
	status code: 400, request id: <masked>
Error: error deleting EventBridge Rule (NTHScheduledChangeRule): ValidationException: Rule can't be deleted since it has targets.
	status code: 400, request id: <masked>
Error: error deleting EventBridge Rule (NTHSpotTermRule): ValidationException: Rule can't be deleted since it has targets.
	status code: 400, request id: <masked>
```

dynamic rule names (e.g. NTHSpotTermRule-<cluster_name>) should do the trick, I can work on it after https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/562

**Comment by bryantbiggs** [comment]
closed in #744
